peepingtom/datablocks/simpleblocks/simpleblock.py

```diff
@@ -37,6 +44,17 @@ def _data_setter(self, data):
         """
         raise NotImplementedError('SimpleBlocks must implement this method')
 
+    @property
+    def reader_function(self):
+        return self._reader_function
+
+    @reader_function.setter
+    def reader_function(self, function):
+        if function is None:
+            self._reader_function = function
+        else:
+            self._reader_function = delayed(function)
```

Comment:
I have a feeling that thise whole `reader_function` interface can be replaced by partials/lambdas. I don't like that you have to pass different things other than `data`.

Reply:
It definitely could be but I'm keen on using Dask for a couple reasons
1. it's already implemented
2. easy to extend to parallelised computation later should it need

There's no extra dependency or anything because napari already depends on dask (or something in the napari stack does)

I tried having an interface like you described where the `DelayedLeaf` object is already aware of everything it needs and it seems like their implementation doesn't like argument-less functions - when I did it that way the function itself got returned from `.compute()` rather than the result of calling the function, it was workable but definitely not how that functionality is intended to be used so I switched to this